ol3-cesium-plugin/gakmlsynchronizer.js

```diff
@@ -0,0 +1,112 @@
+goog.provide('olcs.GaKmlSynchronizer');
+goog.require('olcs.AbstractSynchronizer');
+
+
+/**
+ * Unidirectionally synchronize geoadmin kml layers to Cesium.
+ * @param {!ol.Map} map
+ * @param {!Cesium.Scene} scene
+ * @param {!Cesium.DataSourceCollection} dataSources
+ * @constructor
+ * @extends {olcs.AbstractSynchronizer}
+ * @api
+ * @struct
+ */
+olcs.GaKmlSynchronizer = function(map, scene, dataSources) {
+
+  /**
+   * @protected
+   */
+  this.dataSources_ = dataSources;
+
+  goog.base(this, map, scene);
+};
+goog.inherits(olcs.GaKmlSynchronizer, olcs.AbstractSynchronizer);
+
+
+/**
+ * @inheritDoc
+ */
+olcs.GaKmlSynchronizer.prototype.createSingleLayerCounterparts =
+    function(olLayer) {
+
+  var dsP;
+  var factory = olcs.obj(olLayer)['getCesiumDataSource'];
+  
+  if (factory) {
+    dsP = factory(this.scene);
+  }
+
+  if (!dsP) {
+    /** @type {string} */
+    var url = olcs.obj(olLayer)['url'];
+    
+    if (!(olLayer instanceof ol.layer.Layer) || olLayer.get('type') != 'KML'
+        || !url || /:\/\/public\./.test(url)) {
```

Comment:
because KML from public are modifiable. So the display between 2d and 3d must be up to date.